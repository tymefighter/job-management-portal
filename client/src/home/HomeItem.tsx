import "../styles/HomeItem.scss";

interface HomeItemProps {
    name: string;
    imageUrl: string;
    children: string;
    className?: string
};

function HomeItem({name, children, imageUrl, className}: HomeItemProps) {
    const divClassName = "home-item" + (className ? ` ${className}` : "");
    return (
        <div className={divClassName}>
            <img className="home-item__image" src={imageUrl} alt={name} />
            <h2 className="home-item__name">{name}</h2>
            <p className="home-item__desc">{children}</p>
        </div>
    );
}

export default HomeItem;