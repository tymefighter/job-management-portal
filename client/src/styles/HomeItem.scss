.home-item {
    display: grid;
    grid-template-columns: 3fr 7fr;
    grid-template-areas: 
        "image name"
        "image desc";
    gap: 2rem;
    border: 0.2rem solid black;
    padding: 0 0.5rem 0 0;
}

.home-item--alternate {
    grid-template-columns: 7fr 3fr;
    grid-template-areas: 
        "name image"
        "desc image";
    padding: 0 0 0 0.7rem;
}

.home-item__image {
    grid-area: image;
    width: 100%;
    height: 18rem;
}

.home-item__name {
    grid-area: name;
    font-size: 2rem;
    margin: 2rem 0 0 0;
    padding: 0;
}

.home-item__desc {
    grid-area: desc;
    font-size: 1.2rem;
    padding: 0;
    margin: 0;
}

@media screen and (max-width: 60rem) {
    .home-item {
        grid-template-columns: 1fr;
        grid-template-areas: 
            "image"
            "name"
            "desc";
        padding: 0;
    }

    .home-item__name {
        text-align: center;
    }

    .home-item__desc {
        padding: 0 0.5rem;
    }
}