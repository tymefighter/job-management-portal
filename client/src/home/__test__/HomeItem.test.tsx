import { render } from "@testing-library/react";
import HomeItem from "../HomeItem";

test("HomeItem props and class-list check test", () => {
    const name = "This is a name";
    const imageUrl = "/this/is/an/image/url.jpeg";
    const para = "This is a paragraph of text. It is being used as a test";
    const className = "these are a few classes which are present";

    const element = <HomeItem name={name} imageUrl={imageUrl}
        className={className}>{para}</HomeItem>;

    const {getByAltText, queryByText} = render(element);

    const image = getByAltText(name);
    expect(image.getAttribute("src")).toEqual(imageUrl);

    expect(queryByText(name)).toBeTruthy();
    expect(queryByText(para)).toBeTruthy();
});

test("HomeItem snapshot test", () => {
    const name = "This is a name";
    const imageUrl = "/this/is/an/image/url.jpeg";
    const para = "This is a paragraph of text. It is being used as a test";
    const className = "these are a few classes which are present";

    const element = <HomeItem name={name} imageUrl={imageUrl}
        className={className}>{para}</HomeItem>;

    const {container} = render(element);    

    expect(container.firstChild).toMatchSnapshot("HomeItem Snapshot");
});