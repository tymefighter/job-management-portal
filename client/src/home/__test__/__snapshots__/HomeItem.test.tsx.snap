// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeItem snapshot test: HomeItem Snapshot 1`] = `
<div
  class="home-item these are a few classes which are present"
>
  <img
    alt="This is a name"
    class="home-item__image"
    src="/this/is/an/image/url.jpeg"
  />
  <h2
    class="home-item__name"
  >
    This is a name
  </h2>
  <p
    class="home-item__desc"
  >
    This is a paragraph of text. It is being used as a test
  </p>
</div>
`;
